### ISSUE TITLE
Please give `validator` access to row data

### ISSUE THREAD

**Issue by cheapsteak** [AllenFang/react-bootstrap-table#963]
Validator currently only has access to cell value
It would be useful for when rows have different types of values to be able to use different validators depending on other values in the row

```js
const data = [
  {type: 'name', value: 'John Snow'},
  {type: 'email', value: 'johnsnow@example.com'},
  // ...etc
];
```

```js
const data = [
  {type: 'start_date', value: someDate},
  {type: 'end_date', value: someDateThatShouldBeAfterStartDate},
  // ...etc
];
```

**Comment by AllenFang** [comment]
@cheapsteak, which condition for validator?  row insert, initial load or somewhere?

> Validator currently only has access to cell value

yap, but it only for cell editing and row insert

**Comment by cheapsteak** [comment]
Ah yes, cell editing is the one I was thinking of :)

**Comment by AllenFang** [comment]
@cheapsteak, so you want a validator on row level after cell editing? I cant figure out whats condition we need it, could you please show me a real case, because I think it's enough for cell level validator. thanks

**Comment by cheapsteak** [comment]
Yup!  

I have a table displaying credentials for different services (facebook graph, twitter rest, twitter ads, blubrry, potentially more). Because each user only has one of each, each credential is its own row, with a "type" (facebook/twitter etc), and a "value" (usually a string token). I want to make the cells for the "value"' field editable, and have the validation depend on which "type" of credential it is

**Comment by AllenFang** [comment]
I think `cellEdit.beforeSaveCell` can help you, check [this](https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/cell-edit/cell-edit-hook-table.js#L36) example, `beforeSaveCell` will pass the whole row for you. let me know if you still have any question. thanks

**Comment by cheapsteak** [comment]
Thanks! I am indeed using that function for now, but it seems to silently no-op when my validation fails, but the validator function results in a visual "shake" when the validation fails. Would I be able to replicate that using beforeSaveCell?

**Comment by AllenFang** [comment]
> but it seems to silently no-op when my validation fails

you means you return `false` but it still saving the row successfully?

**Comment by cheapsteak** [comment]
Ah sorry I should clarify
The end result is as expected, they both don't save the row
However, validator "shakes" the input box before hiding the input box (giving a visual hint that the input was not valid), while "beforeSaveCell" just immediately hides the input box

**Comment by AllenFang** [comment]
ok, I got you, if I pass the row data to the validator, is that helpful for you right?  if yes, I can try to support it

**Comment by cheapsteak** [comment]
Yup! That would be great :D

**Comment by AllenFang** [comment]
Released on `v2.10.0`, check this [commit](https://github.com/AllenFang/react-bootstrap-table/commit/474f05de27874c8dfeebc72f471096f334dd0626) and [example](https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/advance/validator-table.js#L32)
